### PR TITLE
Geo searching

### PR DESCRIPTION
**Pull request by lukecampbell**
Adds the ability to make geospatial bounding box searches

Either:

```
discovery.query_geo_bbox(index_id, field, top_left=[x,y], bottom_right=[x2,y2])
```

or using the DSL

```
search_string = "search 'nominal_location' geo box top-left lat 40 lon 0 bottom-right lat 0 lon 40 from 'devices_index'"
discovery.parse(search_string)
```

### CODE REVIEW

File: ion/services/dm/presentation/discovery_service.py [hunk 1]
```diff
@@ -502,6 +541,52 @@ def query_geo_distance(self, source_id='', field='', origin=None, distance='', u
         return self._results_from_response(response,id_only)
 
 
+    def query_geo_bbox(self, source_id='', field='', top_left=None, bottom_right=None, order=None, limit=0, offset=0, id_only=False):
+        validate_true(isinstance(top_left, list) or isinstance(top_left,tuple), 'Top Left is not a list or a tuple')
```
Comment (dstuebe) [inline ion/services/dm/presentation/discovery_service.py]: isinstance takes a tuple of types. Don't use 'or'.

Comment (dstuebe) [inline ion/services/dm/presentation/discovery_service.py]: isinstance(top_left, (list, tuple))

**Comment by lukecampbell** [comment]
Didn't know you could do that.

**Comment by dstuebe** [comment]
Glad I am still good for something!

**Pull request merged**